dualscreeninfo/android/src/main/kotlin/com/microsoft/reactnativedualscreen/ReactNativeDualScreenPackage.kt

```diff
@@ -10,14 +10,25 @@ import com.microsoft.reactnativedualscreen.dualscreen.HingeManager
 
 class ReactNativeDualScreenPackage : ReactPackage {
 	override fun createNativeModules(reactContext: ReactApplicationContext): List<NativeModule> {
-		return listOf(
-				DualScreenInfo(reactContext)
-		)
+		return try {
+			Class.forName("com.microsoft.device.display.DisplayMask", false, this.javaClass.classLoader)
+			listOf(
+					DualScreenInfo(reactContext)
+			)
+		} catch (e: ClassNotFoundException) {
+			listOf()
+		}
```

Comment:
This fix has a big limitation - when on a single screen device, the rotation and windowRects will now be reported as hard-coded (and possibly incorrect) values.  We still need the native module to report the correct orientation and windowRects.  From a quick scan of the code it looks like the native module was trying to gracefully fall back to use data for one screen if the DisplayMask feature isn't present, but it was missing a critical check here:

 private val mDisplayMask: DisplayMask?
  get() {
   return if(currentActivity != null) DisplayMask.fromResourcesRect(currentActivity) else null
  }

It looks like this can be fixed by changing this to:

 private val mDisplayMask: DisplayMask?
  get() {
   return if(currentActivity != null && reactApplicationContext.packageManager.hasSystemFeature(FEATURE_NAME)) DisplayMask.fromResourcesRect(currentActivity) else null
  }

Can you try this alternative fix?

Reply:
Unfortunately that does not work (still ClassNotFoundException). It seems like the class implementation needs to be available if you're going to keep references to it like that? I think it is necessary to use reflection to do what you want to do here.

Reply:
The other alternative being, keep a single screen version of DualScreenInfo and select based on whether or not DisplayMask is available.

Reply:
(p.s. I am not an expert of kotlin or jvm internals so maybe I am overlooking something)

Reply:
@joslevin thanks for trying this out!  I am also not a kotlin expert, there may be a simple solution...  @thewulf7  any ideas here?

Reply:
Here is what it looks like using reflection... IMO this is not a good idea, you lose type safety and make it easier for arbitrary code execution

```kotlin
	private val mDisplayMask: Any?
		get() {
			return try {
				var myClass = Class.forName("com.microsoft.device.display.DisplayMask", true, this.javaClass.classLoader)
				myClass.kotlin.staticFunctions.find { m -> m.name == "fromResourcesRect" }!!.call(currentActivity)
			} catch (e: ClassNotFoundException) {
				return null;
			} catch (e: Throwable){
				return null;
			}
//			return if(currentActivity != null) DisplayMask.fromResourcesRect(currentActivity) else null
		}
	private val rotation: Int
		get() {
			val wm = currentActivity?.getSystemService(Context.WINDOW_SERVICE) as WindowManager?
			return wm?.defaultDisplay?.rotation ?: Surface.ROTATION_0
		}
	private val hinge: Rect
		get() {
//			val boundings: MutableList<Rect>? = mDisplayMask?.getBoundingRectsForRotation(rotation)

			val boundings: MutableList<Rect>? = if (mDisplayMask != null) mDisplayMask!!::class.members.find { m -> m.name == "getBoundingRectsForRotation" } ?.call(mDisplayMask, rotation) as MutableList<Rect>?;
				else null

			return if (boundings == null || boundings.size == 0) {
				Rect(0, 0, 0, 0)
			} else boundings[0]
		}

```

Reply:
Mind you, if this is the only option, I could wrap the reflected part up into a "proxy" class and it would look better. And it does work, and maintains the existing code structure (so no special "single screen" implementation needed)

Reply:
Here is what that would look like: https://github.com/joslevin/react-native-dualscreen/commit/1efade4e64ca3cc0379b67631a0da14f006c412a

Reply:
Yes I like the proxy class, can you get a new revision up with that?

Reply:
Will do 👍